components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/context/model/UserEntity.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.core.context.model;
+
+import org.wso2.carbon.identity.core.context.model.InitiatingEntity;
+
+/**
+ * Entity class for User.
+ * This class holds the authenticated user entity details for a given flow.
+ */
+public class UserEntity implements InitiatingEntity {
+
+    private final String userId;
+    private final String username;
+    private final String loginIdentifier;
```

Comment:
Do we really need these?

Reply:
IMO, at a particular point of time we will only have the user id, loginIdentifier and possibly the login identifier attribute.

Again as this is is not a priority at the moment we can skip adding login identifier and keep only userId, or have userId and loginIdentifier only

Reply:
Fixed only to have userId at this moment.